src/block-cache.ts

```diff
@@ -150,10 +151,11 @@ export function createBlockCacheMiddleware({
 
   // create caching strategies
   const blockCache: BlockCacheStrategy = new BlockCacheStrategy();
-  const strategies: Record<string, BlockCacheStrategy> = {
-    perma: blockCache,
-    block: blockCache,
-    fork: blockCache,
+  const strategies: Record<CacheStrategy, BlockCacheStrategy | undefined> = {
```

Comment:
Does the type need to be specified here? Is it just for explicitness?

Reply:
It was pre-existing; I hadn't really considered whether it was needed or not, I just made the required updates to it. This library is full of unnecessary types.

Reply:
Looks like when I remove this, the inferred type no longer uses `CacheStrategy` as the keys. It gets reduced to the strings instead. Maybe better to preserve the enum as the key so it's clear that it includes all strategies.

Reply:
Interesting. No problem.